### ISSUE TITLE
If the minion Carnivorous Cube eats is transformed, Cube's deathrattle summons the post-transformed minion (or even hero)

### ISSUE THREAD

**Issue by Kybxd** [HearthSim/hs-bugs#1082]
Here is an interesting test showing that Cube's deathrattle summons Frost Lich Jaina. First Cube eats Anub'arak, then play Anub'arak again and Polymorph it. Try to transform it into Chameleos with Recombobulator. Return it to hand and it tranforms into opponent's only card Frost Lich Jaina. Then kill the Cube. 

(Starting from 8:00) https://www.bilibili.com/video/av37117483

**Comment by troggnostupidhs** [comment]
Is it starting at 8:00 or 8:45? I'm not sure what the video is showing that starts at 8:00.

**Comment by Kybxd** [comment]
Well, 8:00 to 8:45 is just showing how to activate this bug, as I described above.

**Comment by troggnostupidhs** [comment]
Reminds me of #984 

And per that bug, this shouldn't happen. Unless it wasn't a general change, maybe?

https://news.blizzard.com/en-us/hearthstone/21964710/update-12-0-august-2-the-boomsday-project

> Whenever a card transforms it loses its enchantments.
>     This change primarily affects the following 4 cards: Shifter Zerus, Molten Blade, Shifting Scroll, and Voodoo Doll. If the target of a Voodoo Doll is transformed or silenced, it is no longer destroyed by the Voodoo Doll’s Deathrattle.

**Comment by Kybxd** [comment]
I guess so. You know when Hearthstone fixes a bug, it's usually in a hard-coded way. #36/#1083 are good examples. In essence, they don't fix a 'bug' thoroughly, but fix a 'case' with public attention.

**Comment by PattuX** [comment]
Jetz and I tried to test this with spells. However, we didn't have Anub'arak but figured out this does not work with Spirit Echo, Redemption, Getaway Kodo, Pyros and also Weasel Tunneler, The Skeleton Knight and Malorne as those create copies. Seems like only Anub'arak works.

edit: After some recombobulating we found that spells are only added to the hand of the cube owner. The zonePos of the cards in hand is the zonePos that the cube had on board.

**Comment by Kybxd** [comment]
Yes, as for deathrattles only Anub'arak is a true zone-moving effect, and other zone-moving deathrattles just create copies. This can be checked with Leyline Manipulator.

**Comment by troggnostupidhs** [comment]
HysteriA's latest video shows this bug

https://youtu.be/Y8kXLhGNmHg?t=1
https://youtu.be/Y8kXLhGNmHg?t=48
https://youtu.be/Y8kXLhGNmHg?t=99
